bin/cleaner/database.js

```diff
@@ -0,0 +1,124 @@
+const { eachSeries, whilst, each } = require('async');
+
+function iterateOverUsers(sqlPool, sqlUsersQuery, onEveryUser, finished) {
+  const chunkSize = 5;
+  let page = 0;
+  let moreResults = true;
+
+  whilst(
+    (cb) => cb(null, moreResults),
+    (cb) => {
+      sqlPool.query(
+        sqlUsersQuery,
+        [chunkSize, page * chunkSize],
+        (err, results) => {
+          if (err) {
+            return cb(err);
+          }
+
+          if (results.length === 0) {
+            moreResults = false;
+
+            return cb();
+          }
+
+          eachSeries(
+            results,
+            onEveryUser,
+            (err) => {
+              if (err) {
+                return cb(err);
+              }
+              page += 1;
+              cb();
+            }
+          );
+        }
+      );
+    },
+    finished
+  );
+}
+
+function getFileCountQuery(sqlPool, sqlFilesQuery, idFile, cb) {
+  sqlPool.query(sqlFilesQuery, [idFile], (err, results) => {
+    if (err) {
+      return cb(err);
+    }
+    if (results.length !== 1) {
+      return cb(new Error('SQL for files didn\'t return a single row'));
+    }
+
+    const count = results[0].count;
+
+    if (typeof count !== 'number') {
+      return cb(new Error('SQL for files didn\'t specify a count numeric column'));
+    }
+
+    cb(null, count);
+  });
+}
+
+const processEntries = (entries, onEveryEntry, cb) => {
+  if (entries.length === 0) {
+    return cb();
+  }
+
+  each(entries, onEveryEntry, cb);
+};
+
+function iterateOverModel(Model, filter, onEveryEntry, cb) {
+  let chunk = [];
+  const chunkSize = 5;
+
+  const cursor = Model.find(filter)
```

Comment:
Better inject the `cursor` directly. As the cursor uses the EventEmitter API, could be replaced for another EventEmitter implementer, also you could avoid coupling things like sorting inside this function. At the same time, you could gain flexibility, allowing the query to be a find with any condition, etc without changing this function

Reply:
Agree, let's change it to:

`iterateOverModel -> iterateOverCursor`